server/client.go

```diff
@@ -452,6 +455,15 @@ func (c *client) processConnect(arg []byte) error {
 			c.authViolation()
 			return ErrAuthorization
 		}
+
+		// Client-enabled tracing. Check permissions if applicable.
+		if c.opts.Trace {
+			if srv.opts.OptInTrace || (c.perms != nil && c.perms.trace) {
+				c.trace = true
+			} else {
+				c.sendErr("Permissions Violation for trace option")
```

Comment:
Adding a return here is causing a failure in the test which I haven't been able to figure out how to fix:

```
--- FAIL: TestUserAuthorizationProto (0.22s)
        test.go:145: Error reading from conn: EOF

                2 - /Users/tylertreat/Go/src/github.com/nats-io/gnatsd/test/user_authorization_test.go:52
                3 - /Users/tylertreat/development/go1.7/src/testing/testing.go:610
                4 - /Users/tylertreat/development/go1.7/src/runtime/asm_amd64.s:2086
```

Any ideas?